fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/CollectionObjectTest.java

```diff
@@ -0,0 +1,140 @@
+		initMocks(this);
+
+		pid = PIDs.get(UUID.randomUUID().toString());
+
+		collection = new CollectionObject(pid, repository, dataLoader);
+
+		folderChildPid = PIDs.get(UUID.randomUUID().toString());
+		when(repository.mintContentPid()).thenReturn(folderChildPid);
+		
+		workChildPid = PIDs.get(UUID.randomUUID().toString());
+		when(repository.mintContentPid()).thenReturn(workChildPid);
+		
+		folderChildObj = new FolderObject(folderChildPid, repository, dataLoader);
+		when(repository.createFolderObject(any(PID.class), any(Model.class)))
+				.thenReturn(folderChildObj);
+		
+		workChildObj = new WorkObject(workChildPid, repository, dataLoader);
+		when(repository.createWorkObject(any(PID.class), any(Model.class)))
+				.thenReturn(workChildObj);
+	}
+
+	@Test
+	public void isValidTypeTest() {
+		// Return the correct RDF types
+		List<String> types = Arrays.asList(PcdmModels.Object.getURI(), Cdr.Collection.getURI());
+		when(dataLoader.loadTypes(eq(collection))).thenAnswer(new Answer<RepositoryObjectDataLoader>() {
+			@Override
+			public RepositoryObjectDataLoader answer(InvocationOnMock invocation) throws Throwable {
+				collection.setTypes(types);
+				return dataLoader;
+			}
+		});
+
+		collection.validateType();
+	}
+	
+	@Test(expected = ObjectTypeMismatchException.class)
+	public void invalidTypeTest() {
+		List<String> types = Arrays.asList(PcdmModels.Object.getURI(), Cdr.Folder.getURI());
+		when(dataLoader.loadTypes(eq(collection))).thenAnswer(new Answer<RepositoryObjectDataLoader>() {
+			@Override
+			public RepositoryObjectDataLoader answer(InvocationOnMock invocation) throws Throwable {
+				collection.setTypes(types);
+				return dataLoader;
+			}
+		});
+
+		collection.validateType();
+	}
+	
+	@Test
+	public void addWorkMemberTest() {
+		collection.addMember(workChildObj);
+
+		ArgumentCaptor<ContentObject> captor = ArgumentCaptor.forClass(ContentObject.class);
+		verify(repository).addMember(eq(collection), captor.capture());
+
+		ContentObject child = captor.getValue();
+		assertTrue("Incorrect type of child added", child instanceof WorkObject);
+		assertEquals("Child did not have the expected pid", workChildPid, child.getPid());
+	}
+
+	@Test
+	public void addFolderMemberTest() {
+		collection.addMember(folderChildObj);
+
+		ArgumentCaptor<ContentObject> captor = ArgumentCaptor.forClass(ContentObject.class);
+		verify(repository).addMember(eq(collection), captor.capture());
+
+		ContentObject child = captor.getValue();
+		assertTrue("Incorrect type of child added", child instanceof FolderObject);
+		assertEquals("Child did not have the expected pid", folderChildPid, child.getPid());
+	}
+
+	// should not be able to add a Collection object as a member
+	@Test(expected = ObjectTypeMismatchException.class)
+	public void addCollectionObjectMemberTest() {
+		CollectionObject childObj = new CollectionObject(workChildPid, repository, dataLoader);
+
+		when(repository.createCollectionObject(any(PID.class), any(Model.class)))
```

Comment:
I can't run the test at the moment, but I don't think this line is doing anything in this test since repository.createCollectionObject is not used in collection.addMember.  The repository create methods would only get called from the CollectionObject class if we had addFolder and addWork methods like with FolderObject, but I'm not clear yet on if we need those, so I'm okay that you left them out of CollectionObject.